frontend/src/SinglePage.js

```diff
@@ -44,27 +44,30 @@ class SinglePage extends React.Component {
   }
 
   async checkSaved() {
-    await fetch(`${backendHostURL}/login/${localStorage.getItem("_id")}`)
-      .then((res) => res.json())
-      .then((data) => {
-        this.setState({ user: this.state.user.concat(data[0]) });
-        console.log("user", this.state.user)
-        if(this.state.user.length > 0)
-        {          
-          const hike = this.state.data._id
-          const length = this.state.user[0].saved_trails.length
-          for(let i = 0; i < length; i++)
-          {
-              const curr = this.state.user[0].saved_trails[i]
-              console.log(curr)
-              if (curr === hike)
-              {
-                this.setState({saved: true})
-                console.log(this.state.saved)
-              }
+    if (localStorage.getItem("isLoggedIn") === "true")
```

Comment:
This is awesome! It looks like this is where you fix the bug of the saved hike. Was this line of checking the local storage for the logged-in user the line that fixed it? 

Also, what does the this.state.user.concat(data[0]) concatenate to?

Reply:
This check was to keep the button as unsaved if you aren't logged in. The bug was fixed by properly adding the saved hikes to the state array and passing it down to the button component !
Because I made this.state.user as an array, this.state.user.concat(data[0]) adds the user info to the state in a readable manner for checking saved hikes - other approaches like this.setState({user: data[0]}) didn't seem to work

Reply:
Nice!

Ah, that makes sense. Yeah, I feel like state is weird sometimes.